zerver/tests/test_markdown.py

```diff
@@ -1156,6 +1158,58 @@ def test_fetch_tweet_data_settings_validation(self) -> None:
         with self.settings(TEST_SUITE=False, TWITTER_CONSUMER_KEY=None):
             self.assertIs(None, fetch_tweet_data("287977969287315459"))
 
+    def mocked_requests_post(*args: Any, **kwargs: Any) -> MockPythonResponse:
+        """This method will be used by the mock to replace requests.post"""
+
+        json_data = kwargs.get("json", {})
+        if not all(json_data.values()):
+            return MockPythonResponse("", 400)
+
+        return MockPythonResponse(args[1], 200)
+
+    def test_kroki_diagrams(self) -> None:
+        def assert_kroki_diagrams(
+            msg: str, expected: str, html: str = "", mock_is_called: bool = True
+        ) -> None:
+            with mock.patch(
+                "requests.post",
+                side_effect=lambda *args, **kwargs: self.mocked_requests_post(
+                    html, *args, **kwargs
+                ),
+            ) as m:
+                self.assertEqual(markdown_convert_wrapper(msg), expected)
+
+                if mock_is_called:
+                    m.assert_called_once()
+
+        # Basic test case.
+        msg = """\n```kroki graphviz\ndigraph G {\nHello->World\n}"""
+        html = """<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"\n "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n<!-- Generated by graphviz version 2.44.0 ()\n -->\n<!-- Title: G Pages: 1 -->\n<svg width="74pt" height="116pt"\n viewBox="0.00 0.00 74.29 116.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">\n<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 112)">\n<title>G</title>\n<polygon fill="white" stroke="transparent" points="-4,4 -4,-112 70.29,-112 70.29,4 -4,4"/>\n<!-- Hello -->\n<g id="node1" class="node">\n<title>Hello</title>\n<ellipse fill="none" stroke="black" cx="33.15" cy="-90" rx="30.59" ry="18"/>\n<text text-anchor="middle" x="33.15" y="-86.3" font-family="Times,serif" font-size="14.00">Hello</text>\n</g>\n<!-- World -->\n<g id="node2" class="node">\n<title>World</title>\n<ellipse fill="none" stroke="black" cx="33.15" cy="-18" rx="33.29" ry="18"/>\n<text text-anchor="middle" x="33.15" y="-14.3" font-family="Times,serif" font-size="14.00">World</text>\n</g>\n<!-- Hello&#45;&gt;World -->\n<g id="edge1" class="edge">\n<title>Hello&#45;&gt;World</title>\n<path fill="none" stroke="black" d="M33.15,-71.7C33.15,-63.98 33.15,-54.71 33.15,-46.11"/>\n<polygon fill="black" stroke="black" points="36.65,-46.1 33.15,-36.1 29.65,-46.1 36.65,-46.1"/>\n</g>\n</g>\n</svg>"""
```

Comment:
These are very long lines; could we read these from a fixture file in `zerver/fixtures/kroki` instead of having them inline here?